Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -54,6 +54,16 @@ const FPSRowData& UPSWorldSubsystem::GetCurrentRow() const
 	return SaveGameInstance->GetCurrentRow();
 }
 
+// Set current row of progression system by tag
+void UPSWorldSubsystem::SetCurrentRowByTag(FPlayerTag NewRowPlayerTag)
+{
+	UPSSaveGameData* SaveGameInstance = GetCurrentSaveGameData();
+	checkf(SaveGameInstance, TEXT("ERROR: 'SaveGameInstanceInternal' is null"));
+
+	SaveGameInstance->SetRowByTag(NewRowPlayerTag);
```

Comment:
Frequent disk writes are heavy in Unreal due to the I/O overhead involved. This can cause noticeable lag on low-spec PC. It's best to avoid writing to save as much as possible: please check first isn't data the same before calling SaveGameInstance

Reply:
solved as part of the https://github.com/h4rdmol/ProgressionSystem/pull/3/commits/6eebcd7f37743666749ea89c7cef79ca579443b2